lib/model/model.go

```diff
@@ -1251,12 +1243,31 @@ func (m *model) ClusterConfig(deviceID protocol.DeviceID, cm protocol.ClusterCon
 	}
 }
 
-// handleIntroductions handles adding devices/shares that are shared by an introducer device
-func (m *model) handleIntroductions(introducerCfg config.DeviceConfiguration, cm protocol.ClusterConfig) (folderDeviceSet, bool) {
+// handleIntroductions handles adding and removing devices/shares that are shared by an introducer device
+func (m *model) handleIntroductions(introducerCfg config.DeviceConfiguration, cm protocol.ClusterConfig) bool {
+	changed := false
+	folders := m.cfg.Folders()
+	devices := m.cfg.Devices()
+
+	defer func() {
```

Comment:
This is too much stuff to keep in the mental stack when reading the method. Should be a named `commitNewFoldersAndDevices(...)` or something... And maybe not deferred but called explicitly on early exit